File: content-solutions/implementation/sim-clock-add.java
import java.util.*;

class Solution {
    public String solution(String time, int minutes) {
        String[] parts = time.split(":");
        int h = Integer.parseInt(parts[0]), m = Integer.parseInt(parts[1]);
        long total = (h * 60L + m + minutes) % 1440;
        return String.format("%02d:%02d", total / 60, total % 60);
    }
}

File: content-solutions/implementation/sim-clock-add.js
function solution(time, minutes) {
  const [h, m] = time.split(":").map(Number);
  const total = (h * 60 + m + minutes) % 1440;
  const hh = String(Math.floor(total / 60)).padStart(2, "0");
  const mm = String(total % 60).padStart(2, "0");
  return `${hh}:${mm}`;
}
